--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -211,7 +211,7 @@
     ELSEIF(IS_APPLE_OS AND CPUINFO_TARGET_PROCESSOR MATCHES "arm64.*")
       LIST(APPEND CPUINFO_SRCS src/arm/mach/init.c)
     ENDIF()
-    IF(CMAKE_SYSTEM_NAME STREQUAL "Android")
+    IF(CMAKE_SYSTEM_NAME STREQUAL "Android" OR CMAKE_C_COMPILER MATCHES "android" OR "$ENV{HOST}" MATCHES "android")
       LIST(APPEND CPUINFO_SRCS
         src/arm/android/properties.c)
     ENDIF()
