diff --git a/configure.ac b/configure.ac
index 1111111..2222222 100644
--- a/configure.ac
+++ b/configure.ac
@@ -947,6 +947,10 @@ AS_IF([test "x$EMSCRIPTEN" = "x"],[
 # include <sys/random.h>
 #endif
 ]], [[
+#ifdef __linux__
+# error getrandom() is currently disabled on Linux to support glibc < 2.25
+#endif
+
 unsigned char buf;
 if (&getrandom != NULL) {
   (void) getrandom((void *) &buf, 1U, 0U);
@@ -971,6 +975,10 @@ AS_IF([test "x$EMSCRIPTEN" = "x"],[
 # include <sys/random.h>
 #endif
 ]], [[
+#ifdef __linux__
+# error getentropy() is currently disabled on Linux to support glibc < 2.25
+#endif
+
 unsigned char buf;
 if (&getentropy != NULL) {
   (void) getentropy((void *) &buf, 1U);
