diff --git a/ndk/checkbuild.py b/ndk/checkbuild.py
index f9ac7e18..6eb99139 100755
--- a/ndk/checkbuild.py
+++ b/ndk/checkbuild.py
@@ -800,7 +800,7 @@ class ShaderTools(ndk.builds.CMakeModule):
         # build fails before the failure that happens in CI, so I can't test a fix for
         # the CI issue. Just disable this until someone that's familiar with the tests
         # has the time to fix them.
-        self.run_ctest = self.host is not Host.Darwin
+        self.run_ctest = False
         super().build()
 
     def install(self) -> None:
